### ISSUE TITLE
Train on custom dataset

### ISSUE THREAD

**Issue by modanesh** [Zehaos/MobileNet#32]
Hi,

I have a custom dataset containing 2 classes, each one has 20,000 image samples. How can I train MobileNet on this dataset?

**Comment by Zehaos** [comment]
Hi,

You should make a tfrecord first. Here is a [example](https://github.com/Zehaos/MobileNet/blob/master/datasets/kitti_object_to_tfrecords.py), and then register it in [dataset_factory](https://github.com/Zehaos/MobileNet/blob/master/datasets/dataset_factory.py).

**Comment by modanesh** [comment]
I've created tfrecord file from my dataset with the following code:

`from random import shuffle
import glob
import cv2
import numpy as np
import tensorflow as tf
import os
import sys


shuffle_data = True  # shuffle the addresses before saving
full_hand_train_path = '/Users/Mohamad/Desktop/MobileNet - RL/MobileNet/full hand/all/*.png'

# read addresses and labels from the 'train' folder
addrs = glob.glob(full_hand_train_path)
labels = [0 if 'fullhand' in addr else 1 for addr in addrs]  # 0 = full hand, 1 = others

# to shuffle data
if shuffle_data:
    c = list(zip(addrs, labels))
    shuffle(c)
    addrs, labels = zip(*c)

# Divide the hata into 60% train, 20% validation, and 20% test
train_addrs = addrs[0:int(0.6*len(addrs))]
train_labels = labels[0:int(0.6*len(labels))]
val_addrs = addrs[int(0.6*len(addrs)):int(0.8*len(addrs))]
val_labels = labels[int(0.6*len(addrs)):int(0.8*len(addrs))]
test_addrs = addrs[int(0.8*len(addrs)):]
test_labels = labels[int(0.8*len(labels)):]



def load_image(addr):
    # read an image and resize to (32, 224)
    # cv2 load images as BGR, convert it to RGB
    img = cv2.imread(addr)
    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
    img = img.astype(np.float32)
    return img


def _int64_feature(value):
  return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))


def _bytes_feature(value):
  return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))


train_filename = '/Users/Mohamad/Desktop/MobileNet - RL/MobileNet/full hand/fullhand_train.tfrecord'  # address to save the TFRecords file

# open the TFRecords file
writer = tf.python_io.TFRecordWriter(train_filename)
for i in range(len(train_addrs)):
    # print how many images are saved every 1000 images
    if not i % 1000:
        print('Train data: {}/{}'.format(i, len(train_addrs)))
        sys.stdout.flush()
    # Load the image
    img = load_image(train_addrs[i])
    label = train_labels[i]
    # Create a feature
    feature = {'train/label': _int64_feature(label),
               'train/image': _bytes_feature(tf.compat.as_bytes(img.tostring()))}
    # Create an example protocol buffer
    example = tf.train.Example(features=tf.train.Features(feature=feature))

    # Serialize to string and write on the file
    writer.write(example.SerializeToString())

writer.close()
sys.stdout.flush()`

**Comment by sinbycos** [comment]
How can I test for image classification on custom dataset? Do I still need to add the custom dataset in the dataset factory?
